rewrite-java-test/src/main/java/org/openrewrite/java/JavaTypeMappingTest.java

```diff
@@ -223,12 +223,16 @@ default void enumTypeB() {
     @Test
     default void ignoreSourceRetentionAnnotations() {
```

Comment:
With the logic changes above this test name no longer reflects what's asserted. These tests seem to have been added by @sambsnyd in https://github.com/openrewrite/rewrite/commit/8f6f78ca to
> "Exclude annotations with Source retention from Java TypeMapping so as to facilitate deduplication of types from different mapping implementations"

Might be good to double check that there's no harm in adding these `RetentionPolicy.SOURCE` annotations now, or restore the original behaviour of skipping those, unless you specifically need them.